examples/simple_saga/src/aggregate.rs

```diff
@@ -0,0 +1,119 @@
+
+        let policies: Vec<Box<dyn SqlitePolicy<LoggingEvent, LoggingError> + Send + Sync>> =
+            vec![Box::new(LoggingPolicy {})];
+
+        SqliteStore::new::<Self>(pool, projectors, policies).await
+    }
+}
+
+impl Identifier for LoggingAggregate {
+    fn name() -> &'static str {
+        MESSAGES
+    }
+}
+
+// A very simply policy, which tries to log an event, and creates another command after it finishes
+// which indicates success or failure to log
+struct LoggingPolicy;
+
+#[async_trait]
+impl SqlitePolicy<LoggingEvent, LoggingError> for LoggingPolicy {
+    async fn handle_event(&self, event: &StoreEvent<LoggingEvent>, pool: &Pool<Sqlite>) -> Result<(), LoggingError> {
+        let id = event.aggregate_id;
+        let agg = LoggingAggregate::new(pool).await?;
+        let state = agg
+            .load(event.aggregate_id)
+            .await
+            .unwrap_or_else(|| AggregateState::new(event.aggregate_id)); // This should never happen
+        match event.payload() {
+            LoggingEvent::Received(msg) => {
+                if msg.contains("fail_policy") {
+                    agg.handle_command(state, LoggingCommand::Fail).await?;
+                    return Err(LoggingError::Domain(msg.clone()));
+                }
+                println!("Logged via policy from {}: {}", id, msg);
+                agg.handle_command(state, LoggingCommand::Succeed).await?;
+            }
+            _ => {}
+        }
+        Ok(())
+    }
+}
+
+#[async_trait]
+impl AggregateManager for LoggingAggregate {
+    type State = u64;
+    type Command = LoggingCommand;
+    type Event = LoggingEvent;
+    type Error = LoggingError;
+
+    fn event_store(&self) -> &(dyn EventStore<Self::Event, Self::Error> + Send + Sync) {
+        &self.event_store
+    }
+
+    fn apply_event(_: &Uuid, state: Self::State, event: &StoreEvent<Self::Event>) -> Self::State {
+        match event.payload() {
+            LoggingEvent::Received(_) => {
+                // Do nothing, since the logging policy is what carries out the side effect
+                // NOTE that, in this case, the Self::State we return isn't valid any more,
+                // since the policy runs another command on this aggregate, causing a state
+                // update which isn't applied to the state returned here
+            }
+            LoggingEvent::Succeeded => {
+                // No-op, we simply persist the event in the event log
+                // This is where some success notification might be sent to another service,
+                // or some other success handling would go, in a more complex system
+            }
+            LoggingEvent::Failed => {
+                // No-op, we simply persist the event in the event log
+                // This is where some cleanup action would go in a more complex system
+            }
+        }
+        state + 1
```

Comment:
I'm using the state here to illustrate the number of events persisted for a given command, so the person reading the example can easily see "ok so when I log a message an event is created to do the logging and then another is created due to the saga policy" etc. I don't mind moving it in to the match, but the idea was just to use it to illustrate the multiple events per command (obviously for the example the aggregate state isn't really necessary)

Reply:
I'm also using it to illustrate how the state returned from handle_command is made invalid by the policy (see comments in main.rs), so always needs to be rebuilt from the DB - moving to inside the match arm for logging wouldn't allow me to do that easily

Reply:
Ok, but why don't give to the state more sense using it in the match..?

Reply:
It's easier to illustrate to potential users about the invalid state if it's a simple count of persisted events, that's why.

Reply:
I am not following the discussion :laughing: what's the problem?

Reply:
I think Simone is asking why I'm using the AggregateState to track the number of events processed, rather than the number of lines logged, and I'm explaining that I'm using it to track the number of events persisted rather than the number of lines logged because it lets me demonstrate a couple of quirks of implementing a saga like this (using a Policy to send the Aggegate another command during the handling of the first one) slightly more straightforwardly.

It's not really a hugely meaningful difference, but I think this way allows me to be clearer in the comments describing what's going on. I'm not sure if/why a difference of opinion this minor is blocking